plugins/utils/neocord.nix

```
@@ -0,0 +1,169 @@
+        Main image display (either "language" or "logo")
+      '';
+
+      client_id = helpers.defaultNullOpts.mkStr "1157438221865717891" ''
+        Use your own Discord application client id. (not recommended)
+      '';
+
+      log_level = helpers.defaultNullOpts.mkEnum ["debug" "info" "warn" "error"] "null" ''
+        Log messages at or above this level.
+      '';
+
+      debounce_timeout = helpers.defaultNullOpts.mkInt 10 ''
+        Number of seconds to debounce events.
+        (or calls to `:lua package.loaded.neocord:update(<filename>, true)`)
+      '';
+
+      enable_line_number = helpers.defaultNullOpts.mkBool false ''
+        Displays the current line number instead of the current project.
+      '';
+
+      blacklist = helpers.defaultNullOpts.mkNullable (types.listOf types.str) "[]" ''
+        A list of strings or Lua patterns that disable Rich Presence if the
+        current file name, path, or workspace matches.
+      '';
+
+      buttons =
+        helpers.defaultNullOpts.mkNullable
+        (types.either helpers.nixvimTypes.rawLua
+          (types.listOf (types.submodule {
+            options = {
+              label = helpers.mkNullOrOption types.str "";
+              url = helpers.mkNullOrOption types.str "";
+            };
+          })))
+        "[]"
+        ''
+          Button configurations which will always appear in Rich Presence.
+          Can be a list of attribute sets, each with the following attributes:
+          `label`: The label of the button. e.g. `"GitHub Profile"`.
+          `url`: The URL the button leads to. e.g. `"https://github.com/<NAME>"`.
+          Can also be a lua function: `function(buffer: string, repo_url: string|nil): table`
+        '';
+
+      file_assets = helpers.mkNullOrOption (with types; attrsOf (listOf str)) ''
+        Custom file asset definitions keyed by file names and extensions.
+        List elements for each attribute (filetype):
+        `name`: The name of the asset shown as the title of the file in Discord.
+        `source`: The source of the asset, either an art asset key or the URL of an image asset.
+        Example:
+        ```nix
+          {
+            # Use art assets uploaded in Discord application for the configured client id
+            js = [ "JavaScript" "javascript" ];
+            ts = [ "TypeScript" "typescript" ];
+            # Use image URLs
+            rs = [ "Rust" "https://www.rust-lang.org/logos/rust-logo-512x512.png" ];
+            go = [ "Go" "https://go.dev/blog/go-brand/Go-Logo/PNG/Go-Logo_Aqua.png" ];
+          };
+        ```
+      '';
+
+      show_time = helpers.defaultNullOpts.mkBool true "Show the timer.";
+
+      global_timer = helpers.defaultNullOpts.mkBool false "if set true, timer won't update when any event are triggered.";
+
+      # Rich presence text options.
+      editing_text = helpers.defaultNullOpts.mkNullable (types.either types.str helpers.nixvimTypes.rawLua) "Editing %s" ''
```

```suggestion
      editing_text = helpers.defaultNullOpts.mkStr "Editing %s" ''
```

`rawType` is accepted by default: you have nothing to do :)
See [here](https://github.com/nix-community/nixvim/blob/22351f4628ddd1556c2a35b5ca9b70405f1626ac/lib/options.nix#L138) (`maybeRaw`).

---

I had no idea, thanks for sharing :+1: